### PR TITLE
[SR-013] - Static Map View

### PR DESCRIPTION
**Pull request by marha333**
#13 
Implemented Static Map View in RouteDetailsFragment

[MH, DK]

### CODE REVIEW

File: app/src/main/java/at/team30/setroute/ui/route_detail/RouteDetailFragment.kt [hunk 1]
```diff
@@ -31,11 +33,25 @@ class RouteDetailFragment : Fragment() {
         super.onViewCreated(view, savedInstanceState)
 
         val route = viewModel.getRoute(args.routeId)
+
+        val imageView = view.findViewById<ImageView>(R.id.static_map_view);
+        val headerView = view.findViewById<ImageView>(R.id.header)
```
Comment (jerlacher) [inline app/src/main/java/at/team30/setroute/ui/route_detail/RouteDetailFragment.kt:46]: We should probably remove the old `headerView` and replace it with some loading indicator.

File: app/src/main/java/at/team30/setroute/ui/route_detail/RouteDetailViewModel.kt [hunk 2]
```diff
@@ -1,16 +1,63 @@
...
         return routesRepository.getRoutesById(id)
     }
+
+    fun requestImage(route: Route?,
+                     responseListener: Response.Listener<android.graphics.Bitmap>,
+                     errorListener: Response.ErrorListener) {
+        if (route == null) {
+            return;
+        }
+        val queue = Volley.newRequestQueue(context)
```
Comment (jerlacher) [inline app/src/main/java/at/team30/setroute/ui/route_detail/RouteDetailViewModel.kt]: Could move the queue to an private member of the ViewModel -> can be reused for other requests in the future.

File: app/src/main/res/layout/fragment_route_detail.xml [hunk 3]
```diff
@@ -19,7 +19,15 @@
                 android:scaleType="centerCrop"
                 android:src="@drawable/ic_sample_layer"
                 android:background="@color/ic_sample_background_background"
-                tools:ignore="ContentDescription" />
+                tools:ignore="ContentDescription"
+                android:visibility="visible"/>
```
Comment (jerlacher) [inline app/src/main/res/layout/fragment_route_detail.xml:23]: Replace header with some loading indicator to give the user feedback that something is actually loading here.

Maybe one of these: https://github.com/ybq/Android-SpinKit  ?

**Review by jerlacher — COMMENTED** [review]
Just some small things we could maybe include, but looks great :+1:

File: app/src/main/java/at/team30/setroute/ui/route_detail/RouteDetailViewModel.kt (hunk 2 above)
Comment (marha333) [reply app/src/main/java/at/team30/setroute/ui/route_detail/RouteDetailViewModel.kt]: Sounds like a good idea :)

File: app/src/main/res/layout/fragment_route_detail.xml (hunk 3 above)
Comment (marha333) [reply app/src/main/res/layout/fragment_route_detail.xml:23]: We should talk with Darja about this and maybe create a new Issue for the loading indicator.

Comment (marha333) [reply app/src/main/res/layout/fragment_route_detail.xml:23]: Oh sorry, looks easy with android spinkit, maybe we don't need another issue

File: app/src/main/java/at/team30/setroute/ui/route_detail/RouteDetailViewModel.kt [hunk 4]
```diff
@@ -1,16 +1,77 @@
...
+                val user = imageRepository.getImage(1) = withContext(Dispatchers.IO);
+                _image.value = user
+
+        }
+    }
+
+
+     */
+    private lateinit var image: LiveData<Bitmap?>
+   // = liveData {
```
Comment (jerlacher) [inline app/src/main/java/at/team30/setroute/ui/route_detail/RouteDetailViewModel.kt:45]: remove unused code ?

File: app/src/main/java/at/team30/setroute/ui/route_detail/RouteDetailViewModel.kt [hunk 5]
```diff
@@ -1,16 +1,77 @@
...
 @HiltViewModel
 class RouteDetailViewModel @Inject constructor(
-        private val routesRepository : IRoutesRepository
+    private val routesRepository : IRoutesRepository,
+    private val imageRepository : IImageRepository
 ) : ViewModel() {
+
+    var routeId = 0;
+
+    /*
```
Comment (jerlacher) [inline app/src/main/java/at/team30/setroute/ui/route_detail/RouteDetailViewModel.kt:28]: unused ?

File: app/src/main/java/at/team30/setroute/infrastructure/ImageRepository.kt [hunk 6]
```diff
@@ -20,29 +22,28 @@ class ImageRepository @Inject constructor( private val routesRepository : IRoute
...
-                                /*
-                        for ((name, value) in response.headers) {
-                            println("$name: $value")
-                        }
-                                 */
-                         image = BitmapFactory.decodeStream(response.body!!.byteStream())
-                    }
+            var response = client.newCall(request).execute()
+
+            if (!response.isSuccessful) throw IOException("Unexpected code $response")
```
Comment (jerlacher) [inline app/src/main/java/at/team30/setroute/infrastructure/ImageRepository.kt:39]: I think this Exception is never called so probably don't throw it. We have a fallback-image anyways

just return null if possible :)

Comment (jerlacher) [reply app/src/main/java/at/team30/setroute/infrastructure/ImageRepository.kt:39]: Nevermind, just saw that the exception is caught. My bad :)

**Review by jerlacher — APPROVED** [review]
I suggest removing the unused code, but otherwise LGTM 👍

**Pull request merged**